src/sequence_state.cc

```diff
@@ -118,23 +152,42 @@ SequenceStates::Initialize(
       }
     }
 
-    std::shared_ptr<AllocatedMemory> data;
+    std::shared_ptr<MutableMemory> data;
     auto initial_state_it = initial_state.find(state_config.input_name());
     if (initial_state_it != initial_state.end()) {
-      data = std::make_shared<AllocatedMemory>(
-          initial_state_it->second.data_->TotalByteSize(),
-          TRITONSERVER_MEMORY_CPU, 0);
-
-      TRITONSERVER_MemoryType memory_type;
-      int64_t memory_type_id;
-      char* dst_buffer = data->MutableBuffer(&memory_type, &memory_type_id);
-      char* initial_state_buffer =
-          initial_state_it->second.data_->MutableBuffer(
-              &memory_type, &memory_type_id);
-
-      memcpy(
-          dst_buffer, initial_state_buffer,
-          initial_state_it->second.data_->TotalByteSize());
+      if (use_growable_memory) {
+        std::unique_ptr<GrowableMemory> growable_memory;
+        RETURN_IF_ERROR(GrowableMemory::Create(
+            growable_memory, initial_state_it->second.data_->TotalByteSize(),
+            TRITONSERVER_MEMORY_GPU, device_id,
+            cuda_virtual_address_size.at(device_id)));
+        data = std::move(growable_memory);
```

Comment:
I'm a little confused at what is happening here:

1) we move the growable_memory to data (this implies that the contents of growable_memory are now moved to data) 

2) we get the buffer from data and call this dst_buffer

3) we copy from the dst_buffer into the growable_memory MutableBuffer()

Isn't dst_buffer and growable_memory.MutableBuffer() the same (since we moved growable_buffer into data?)

Reply:
great catch! Fixed.